File: anb/leetcode/linked list/206. Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        val_list = []
        cur = head
        while cur != None:
            val_list.append(cur.val)
            cur = cur.next
        
        cur = head
        for i in reversed(range(len(val_list))):
            cur.val = val_list[i]
            cur = cur.next
        return head

File: 100milliongold/leetcode/linked-list/206.Reverse Linked List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
 var reverseList = function(head) {
    let answer = null, point = head;
    
    
    while(point){
        let temp = point.next;
        point.next = answer;
        answer = point;
        point = temp
    }
    
    
    return answer
};